test/unit/suite.spec.js

```diff
@@ -545,6 +545,89 @@ describe('Suite', function() {
       expect(suite.timeout(), 'to be', 100);
     });
   });
+
+  describe('hasOnly()', function() {
+    it('should return true if a test has `only`', function() {
+      var suite = new Suite('foo');
+      var test = new Test('bar');
+
+      suite.appendOnlyTest(test);
+
+      expect(suite.hasOnly(), 'to be', true);
+    });
+
+    it('should return true if a suite has `only`', function() {
+      var suite = new Suite('foo');
+      var nested = new Suite('bar');
+
+      suite.appendOnlySuite(nested);
+
+      expect(suite.hasOnly(), 'to be', true);
+    });
+
+    it('should return true if nested suite has `only`', function() {
+      var suite = new Suite('foo');
+      var nested = new Suite('bar');
+      var test = new Test('baz');
+
+      nested.appendOnlyTest(test);
+      // `nested` has a `only` test, but `suite` doesn't know about it
+      suite.suites.push(nested);
+
+      expect(suite.hasOnly(), 'to be', true);
+    });
+
+    it('should return false if no suite or test is marked `only`', function() {
+      var suite = new Suite('foo');
+      var nested = new Suite('bar');
+      var test = new Test('baz');
+
+      suite.suites.push(nested);
+      nested.tests.push(test);
+
+      expect(suite.hasOnly(), 'to be', false);
+    });
+  });
+
+  describe('.filterOnly()', function() {
+    it('should filter out all other tests and suites if a test has `only`', function() {
+      var suite = new Suite('a');
+      var nested = new Suite('b');
+      var test = new Test('c');
+      var test2 = new Test('d');
+
+      suite.suites.push(nested);
+      suite.appendOnlyTest(test);
+      suite.tests.push(test2);
+
+      suite.filterOnly();
+
+      expect(suite.suites.length, 'to be', 0);
```

Comment:
it's best to make a single assertion where possible.  you can do this

```js
expect(suite, 'to satisfy', {
  suites: expect.it('to be empty'),
  tests: expect.it('to have length', 1).and('to contain', {title: 'c'})
});
```

similarly below

Reply:
@boneskull I addressed the comments, but I'm curious: why is a single assertion best?

Reply:
I myself prefer multiple explicit assertions that I don't have to lookup within unexpected config files. Seems like violation of the KISS mentality...

Reply:
TBH I also prefer multiple explicit assertions, but that's mostly because I'm not very good with unexpected's syntax off the top of my head :)